File: src/main/java/com/wlyscraft/slimefunitemexpansion/items/HopeFlower.java
package com.wlyscraft.slimefunitemexpansion.items;

import io.github.thebusybiscuit.slimefun4.api.events.PlayerRightClickEvent;
import io.github.thebusybiscuit.slimefun4.api.items.ItemSetting;
import io.github.thebusybiscuit.slimefun4.api.items.settings.IntRangeSetting;
import io.github.thebusybiscuit.slimefun4.core.attributes.NotPlaceable;
import io.github.thebusybiscuit.slimefun4.core.attributes.RandomMobDrop;
import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
import me.mrCookieSlime.Slimefun.Lists.RecipeType;
import me.mrCookieSlime.Slimefun.Objects.Category;
import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
import org.bukkit.inventory.ItemStack;

import javax.annotation.Nonnull;

public class HopeFlower extends SimpleSlimefunItem<ItemUseHandler> implements NotPlaceable, RandomMobDrop {
    private final ItemSetting<Boolean> dropSetting = new ItemSetting<>("drop-from-players", true);
    private final ItemSetting<Integer> chance = new IntRangeSetting("players-drop-chance", 0, 5, 100);

    public HopeFlower(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(category, item, recipeType, recipe);

        addItemSetting(dropSetting);
        addItemSetting(chance);
    }

    @Override
    public int getMobDropChance() {
        return chance.getValue();
    }

    public boolean isDroppedFromPlayers() {
        return dropSetting.getValue();
    }

    @Nonnull
    @Override
    public ItemUseHandler getItemHandler() {
        return PlayerRightClickEvent::cancel;
    }
}

File: src/main/kotlin/io/github/itsflicker/slimefunitemexpansion/items/HopeFlower.kt
package io.github.itsflicker.slimefunitemexpansion.items

import io.github.thebusybiscuit.slimefun4.api.events.PlayerRightClickEvent
import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.ItemSetting
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.items.settings.IntRangeSetting
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import io.github.thebusybiscuit.slimefun4.core.attributes.NotPlaceable
import io.github.thebusybiscuit.slimefun4.core.attributes.RandomMobDrop
import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler
import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem
import org.bukkit.inventory.ItemStack

class HopeFlower(itemGroup: ItemGroup, item: SlimefunItemStack, recipeType: RecipeType, recipe: Array<ItemStack?>) :
    SimpleSlimefunItem<ItemUseHandler>(itemGroup, item, recipeType, recipe), NotPlaceable, RandomMobDrop {

    private val dropSetting = ItemSetting(this, "drop-from-villagers", true)
    private val chance: ItemSetting<Int> = IntRangeSetting(this, "villagers-drop-chance", 0, 5, 100)

    override fun getMobDropChance(): Int {
        return chance.value
    }

    val isDroppedFromPlayers: Boolean
        get() = dropSetting.value

    override fun getItemHandler(): ItemUseHandler {
        return ItemUseHandler { obj: PlayerRightClickEvent -> obj.cancel() }
    }

    init {
        addItemSetting(dropSetting)
        addItemSetting(chance)
    }
}